### DISCUSSION TITLE
Typst - IEEE - abstract - generalise layout

### DISCUSSION THREAD

**Discussion by rainer-rq-koelle** [quarto-dev/quarto-cli#8639] [Feature Requests]
### Description

Updated Quarto and RStudio today to play around with Typst and/or decide whether to give it a try for an upcoming conference paper submission. Thanks for all the hard work on Typst, it really feels lightweight.
From a youtube video of posit::2023, I understood that format observations should also be posted here. If I got this wrong, please kill this "request" (and let me know where to put it instead). 

The bit I noted: The current typst-ieee format prints the abstract across the whole page. Typically, in IEEE conference proceedings, the abstract sits in the column layout. It feels like the column settings only take effect after the abstract section.

For the case at hand (IEEE conference proceedings requirement) the abstract should be part of the column layout of the document.
However, I could picture for own layouts that the author would like to have the abstract spanning (similar to the title) the full body of the page. 
Ditto for a table-of-contents.

The idea would be to have this controlled via a key like _abstract-in-column_ (or _toc-in-column_).

For the typst-ieee format this could be automatically set to _true_ (as IEEE requires it).

Next to the number of columns, it would be nice to be able to control - next to the page margins - also the gap between the columns (e.g. _column-spacing_).

Again, thanks for all the hard work on **typst** and helping non-LATEXer starting to tame the pdf landscape.
[q14-ieee-typst.pdf](https://github.com/quarto-dev/quarto-cli/files/14199562/q14-ieee-typst.pdf)

**Comment by mcanouil** [comment]
Thanks for the feedback.
I think it would make more sense for this to go where the templates live rather than in Quarto CLI: <https://github.com/quarto-ext/typst-templates>.
cc @cderv

**Comment by rainer-rq-koelle** [comment]
Thanks for that pointer. Will put it there and close here. Servus, RQ

**Reply by mcanouil** [reply]
For reference:
- https://github.com/quarto-ext/typst-templates/issues/10
